lib/node-static.js

```diff
@@ -137,8 +137,9 @@ Server.prototype.servePath = function (pathname, status, headers, req, res, fini
     pathname = this.resolve(pathname);
 
     // Make sure we're not trying to access a
-    // file outside of the root.
-    if (pathname.indexOf(that.root) === 0) {
+    // file outside of the root or an invalid
+    // pathname.
+    if (pathname.indexOf(that.root) === 0 && pathname.indexOf('\u0000') === -1) {
```

Comment:
You don't log paths like `../` either; I just modeled my code after the existing code.

I'm not aware of documentation of this problem. I became aware of it because someone attacked my server with it.